### PR TITLE
feat: geo 1012 implement announcement file upload

### PR DESCRIPTION
**Pull request by goemen**
# Description

Implement announcement file upload

Fixes # ([issue](https://finrms.atlassian.net/browse/GEO-1012))

## Type of change

Please delete options that are not relevant.
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://pay-transparency-pr-662-frontend.apps.silver.devops.gov.bc.ca)
- [Admin Frontend](https://pay-transparency-pr-662-admin-frontend.apps.silver.devops.gov.bc.ca)
- [Backend external API console](https://pay-transparency-pr-662-backend-external.apps.silver.devops.gov.bc.ca/api/V1/docs)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/fin-pay-transparency/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/fin-pay-transparency/actions/workflows/merge.yml)

### CODE REVIEW

File: backend/src/v1/middlewares/storage/upload.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      const s3 = new S3Client({
+        credentials: {
+          accessKeyId,
+          secretAccessKey,
+        },
+        endpoint,
+        forcePathStyle: true,
+        region,
+      });
+      const stream = fs.createReadStream(path);
```
[the remark this answers is not in the document]
Comment (goemen) [reply backend/src/v1/middlewares/storage/upload.ts:48]: This is a false positive, the path points to the os temp folder if there is a file upload. Users do not control this path

File: admin-frontend/src/components/EditAnnouncementPage.vue [hunk 2]
```diff
@@ -29,6 +31,7 @@ const submit = async (data: AnnouncementFormValue) => {
     NotificationService.pushNotificationSuccess(
       'Announcement saved successfully',
     );
+    await announcementSearch.repeatSearch();
     router.push('/announcements');
   } catch (error) {
     console.error('Failed to save announcement', error);
```
Comment (sukanya-rath) [inline admin-frontend/src/components/EditAnnouncementPage.vue:37]: log.error

File: admin-frontend/src/services/apiService.ts [hunk 3]
```diff
@@ -340,17 +340,24 @@ export default {
...
   async addAnnouncement(data: AnnouncementFormValue) {
+    const formData = buildFormData(data);
+
     try {
-      return await apiAxios.post(ApiRoutes.ANNOUNCEMENTS, data);
+      return await apiAxios.post(ApiRoutes.ANNOUNCEMENTS, formData, {
+        headers: { 'Content-Type': 'multipart/form-data' },
+      });
     } catch (e) {
       console.log(`Failed to post from Nodejs addAnnouncement API - ${e}`);
```
Comment (sukanya-rath) [inline admin-frontend/src/services/apiService.ts:350]: Update all the console.log and error

**Review by sukanya-rath — CHANGES_REQUESTED** [review]

File: admin-frontend/src/components/EditAnnouncementPage.vue (hunk 2 above)
Comment (goemen) [reply admin-frontend/src/components/EditAnnouncementPage.vue:37]: The frontend uses console.error

File: admin-frontend/src/services/apiService.ts (hunk 3 above)
Comment (goemen) [reply admin-frontend/src/services/apiService.ts:350]: Do you mean change from console.log to console.error? @sukanya-rath

**Review by sukanya-rath — APPROVED** [review]

**Pull request merged**